data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -175,67 +173,110 @@ private void overwriteExistingData() throws IOException {
     DataFile file12 =
-        writeFile(sharedTableLocation, format.addExtension("file-12"), file1SecondSnapshotRecords);
+        writeFile(
+            sharedTableLocation, fileFormat.addExtension("file-12"), file1SecondSnapshotRecords);
     DataFile file22 =
-        writeFile(sharedTableLocation, format.addExtension("file-22"), file2SecondSnapshotRecords);
+        writeFile(
+            sharedTableLocation, fileFormat.addExtension("file-22"), file2SecondSnapshotRecords);
     DataFile file32 =
-        writeFile(sharedTableLocation, format.addExtension("file-32"), file3SecondSnapshotRecords);
+        writeFile(
+            sharedTableLocation, fileFormat.addExtension("file-32"), file3SecondSnapshotRecords);
 
     sharedTable.newFastAppend().appendFile(file12).appendFile(file22).appendFile(file32).commit();
 
     DataFile file13 =
-        writeFile(sharedTableLocation, format.addExtension("file-13"), file1ThirdSnapshotRecords);
+        writeFile(
+            sharedTableLocation, fileFormat.addExtension("file-13"), file1ThirdSnapshotRecords);
     DataFile file23 =
-        writeFile(sharedTableLocation, format.addExtension("file-23"), file2ThirdSnapshotRecords);
+        writeFile(
+            sharedTableLocation, fileFormat.addExtension("file-23"), file2ThirdSnapshotRecords);
     DataFile file33 =
-        writeFile(sharedTableLocation, format.addExtension("file-33"), file3ThirdSnapshotRecords);
+        writeFile(
+            sharedTableLocation, fileFormat.addExtension("file-33"), file3ThirdSnapshotRecords);
 
     sharedTable.newFastAppend().appendFile(file13).appendFile(file23).appendFile(file33).commit();
   }
 
-  @Before
+  @BeforeEach
   public void createTables() throws IOException {
-    File location = temp.newFolder("shared");
-    Assert.assertTrue(location.delete());
+    File location = temp;
+    Assertions.assertThat(location.delete()).isTrue();
     this.sharedTableLocation = location.toString();
     this.sharedTable =
         TABLES.create(
             SCHEMA,
             PartitionSpec.unpartitioned(),
-            ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, format.name()),
+            ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, temp.getName()),
             sharedTableLocation);
 
     Record record = GenericRecord.create(SCHEMA);
 
-    DataFile file1 =
-        writeFile(sharedTableLocation, format.addExtension("file-1"), file1FirstSnapshotRecords);
-
-    Record nullData = record.copy();
-    nullData.setField("id", 11L);
-    nullData.setField("data", null);
-
-    DataFile file2 =
-        writeFile(sharedTableLocation, format.addExtension("file-2"), file2FirstSnapshotRecords);
-    DataFile file3 =
-        writeFile(sharedTableLocation, format.addExtension("file-3"), file3FirstSnapshotRecords);
-
-    // commit the test data
-    sharedTable.newAppend().appendFile(file1).appendFile(file2).appendFile(file3).commit();
+    String[] format = {"parquet", "orc", "avro"};
```

Comment:
this isn't correct. As I mentioned further above, the class should be a parameterized test, thus these changes can all be reverted here